vertical-pod-autoscaler/e2e/v1/recommender.go

```diff
@@ -226,6 +226,73 @@ var _ = RecommenderE2eDescribe("VPA CRD object", func() {
 	})
 })
 
+var _ = RecommenderE2eDescribe("VPA ContainerScalingMode", func() {
+	f := framework.NewDefaultFramework("vertical-pod-autoscaling")
+
+	var (
+		vpaClientSet *vpa_clientset.Clientset
+	)
+
+	ginkgo.BeforeEach(func() {
+		config, err := framework.LoadConfig()
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		vpaClientSet = vpa_clientset.NewForConfigOrDie(config)
+	})
+
+	ginkgo.It("ContainerScalingModeAuto recommendations", func() {
+		ginkgo.By("Setting up a hamster deployment")
+		createTwoHamestersDeployment(f)
+		vpaCRD := createVpaCRDWithContainerScalingMode(f, vpa_types.ContainerScalingModeAuto)
+
+		ginkgo.By("Waiting for recommendation to be filled")
+		vpa, err := WaitForRecommendationPresent(vpaClientSet, vpaCRD)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(len(vpa.Status.Recommendation.ContainerRecommendations)).Should(gomega.BeNumerically("==", 2))
+
+	})
+
+	ginkgo.It("ContainerScalingModeOff recommendations", func() {
+		ginkgo.By("Setting up a hamster deployment")
+		createTwoHamestersDeployment(f)
+		vpaCRD := createVpaCRDWithContainerScalingMode(f, vpa_types.ContainerScalingModeOff)
+
+		ginkgo.By("Waiting for recommendation to be filled")
+		vpa, err := WaitForRecommendationPresent(vpaClientSet, vpaCRD)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		// hamster container has recommendations turned off,
+		// so we expect only one container recommendation for hamster2.
+		gomega.Expect(len(vpa.Status.Recommendation.ContainerRecommendations)).Should(gomega.BeNumerically("==", 1))
+		gomega.Expect(vpa.Status.Recommendation.ContainerRecommendations[0].ContainerName).To(gomega.Equal("hamster2"))
+	})
+})
+
+func createTwoHamestersDeployment(f *framework.Framework) {
+	d := NewHamsterDeploymentWithResourcesAndLimits(f,
+		ParseQuantityOrDie("300m") /*cpu request*/, ParseQuantityOrDie("300Mi"), /*memory request*/
+		ParseQuantityOrDie("300m") /*cpu limit*/, ParseQuantityOrDie("300Mi") /*memory limit*/)
+	d.Spec.Template.Spec.Containers = append(d.Spec.Template.Spec.Containers, d.Spec.Template.Spec.Containers[0])
+	d.Spec.Template.Spec.Containers[1].Name = "hamster2"
+
+	_ = startDeploymentPods(f, d)
+}
+
+func createVpaCRDWithContainerScalingMode(f *framework.Framework, mode vpa_types.ContainerScalingMode) *vpa_types.VerticalPodAutoscaler {
+	vpaCRD := NewVPA(f, "hamster-vpa", hamsterTargetRef)
+	updateMode := vpa_types.UpdateModeAuto
```

Comment:
You don't need to set UpdateMode for this test since this just tests recommender

Reply:
Removed.